### ISSUE TITLE
Suggest import nearest current file.

### ISSUE THREAD

**Issue by ryansmith94** [Microsoft/TypeScript#28619]
## Search Terms
vs code, language service, suggestions, auto complete, import, symbol, nearest current file.

## Suggestion
In VS Code the auto complete dropdown for auto importing seems to have a random order. I think the user experience could be improved by ordering the exported symbols in the dropdown in order of how close the file path is to your current file path.

## Use Cases/Examples
See the example directory structure below.

- moduleA/
  - constants.ts
  - index.ts
- moduleB/
  - constants.ts
  - index.ts

Say that both moduleA/constants.ts and moduleB/constants.ts export a variable like `foobar`. When I type `foobar` in moduleA/index.ts it's likely that I'm trying to auto import `foobar` from moduleA/constants.ts`.

In the current approach the dropdown suggests `foobar` from both moduleA/constants.ts and moduleB/constants.ts, however, the order doesn't seem to be guaranteed so sometimes it will suggest moduleB's `foobar` first and other times moduleA's `foobar`.

I think this could be improved by ordering suggestions in terms of how close their source file path is to the current file path, such that moduleA's `foobar` always appears before moduleB's `foobar` when I'm in moduleA/index.ts. For the sake of completeness, I think it should show moduleB's `foobar` first when I'm in moduleB/index.ts.

I hope I've explained that clearly enough.

## Checklist

My suggestion meets these guidelines:

* [X] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [X] This wouldn't change the runtime behavior of existing JavaScript code
* [X] This could be implemented without emitting different JS based on the types of the expressions
* [X] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [X] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by AshleyGrant** [comment]
I agree, this is currently very frustrating for me. In a project I'm working on, we utilize a state management library, and end up having many functions that share the same name in many different modules, but the function in the closest module is the one I want 99.99% of the time. Unfortunately, the language service always seems to suggest the modules I don't want, and the closest module is buried at the end of the list quite often. Looking at the picture below, no human would order the suggestions that way. 

![image](https://user-images.githubusercontent.com/9885322/49121971-896a9780-f280-11e8-8146-011a46d61dab.png)

**Comment by weswigham** [comment]
Order isn't random - it's nonrelative first, then relative. Within those, they're sorted by number of path segments (and then lexicographically within that). Generally speaking, in many cases, the import directly from a package (vs some deep, relative import) is usually correct; but I assume you're using the `baseUrl` compiler option to give a bunch of your files nonrelative names?

**Comment by be5invis** [comment]
Really want this. TS' auto import sometimes even introduce cyclic dependencies, if a directory re-exports stuff within `index.ts`.

**Comment by ivosabev** [comment]
I am not sure why this is in the Backlog, this is a HUGE time waster, node_module packages are always listed first, instead of your own file. Import a packages like aws and you have to scroll a lot to reach your own files.

<img width="509" alt="Screen Shot 2022-02-05 at 2 46 13 PM" src="https://user-images.githubusercontent.com/103370/152642715-247562b0-1440-4780-8fc9-73a2a7573ccf.png">

**Comment by RyanCavanaugh** [comment]
This is implemented under the "import module specifier" -> "shortest" setting now
